gobblin-salesforce/src/main/java/org/apache/gobblin/salesforce/SalesforceSource.java

```diff
@@ -78,9 +79,23 @@
   public static final boolean DEFAULT_USE_ALL_OBJECTS = false;
 
   private static final String ENABLE_DYNAMIC_PARTITIONING = "salesforce.enableDynamicPartitioning";
-  private static final String DAY_PARTITION_QUERY_TEMPLATE = "SELECT count(${column}) cnt, DAY_ONLY(${column}) time FROM ${table} "
-      + "WHERE ${column} ${greater} ${start} AND ${column} ${less} ${end} GROUP BY DAY_ONLY(${column}) ORDER BY DAY_ONLY(${column})";
-  private static final String DAY_FORMAT = "yyyy-MM-dd";
+  private static final String ENABLE_DYNAMIC_PROBING = "salesforce.enableDynamicProbing";
+  private static final String DYNAMIC_PROBING_LIMIT = "salesforce.dynamicProbingLimit";
+  private static final int DEFAULT_DYNAMIC_PROBING_LIMIT = 1000;
+  private static final String MIN_TARGET_PARTITION_SIZE = "salesforce.minTargetPartitionSize";
+  private static final int DEFAULT_MIN_TARGET_PARTITION_SIZE = 250000;
+  private static final String PROBE_TARGET_RATIO = "salesforce.probeTargetRatio";
+  private static final double DEFAULT_PROBE_TARGET_RATIO = 0.60;
```

Comment:
This is a great idea!  We should document it so that readers can understand its meaning.